package/gluon-core/luasrc/usr/lib/lua/gluon/site.lua

```diff
@@ -16,11 +15,40 @@ local site = (function()
 	file:close()
 
 	return assert(decoder:get())
-end)()
+end
+
+local site = read_json('/lib/gluon/site.json')
+local domain = (function(site)
+	local uci = require('simple-uci').cursor()
+	local fs = require "nixio.fs"
+	local sname = uci:get_first('gluon', 'system')
+
+	local dn = uci:get('gluon', sname, 'domain') or ''
+
+	if fs.stat('/lib/gluon/domains/'..dn..'.json', 'type')~='reg' then
+		dn = site['default_domain']
+	end
+
+	local domain = read_json('/lib/gluon/domains/'..dn..'.json')
+	if domain['domain_aliases'] and domain['domain_aliases'][dn] then
```

Comment:
This currently makes not much sense to me. The file is already `..dn..'.json'`, why can't you just set the `domain_name` there?

Reply:
Aliases for domains are created with symlinks. So in the symliked case `domain_name` is incorrect in the json file and has to be replaced with `domain['domain_aliases'][dn]`.